### PR TITLE
Dep groupby corrwith

### LINKED ISSUE #57158: DEPR: groupby.corrwith
As far as I can tell, [corrwith was added](https://github.com/pandas-dev/pandas/pull/5604) simply because it could be along with other methods on DataFrame. [Searching SO](https://stackoverflow.com/search?q=%5Bpandas%5D+groupby+%22corrwith%22) gives two results, of which `corrwith` is not involved in the solution for both. I'm guessing this method really doesn't see much if any use.

It's signature is odd - it pairs each of the groups up with a single DataFrame and aligns them.

```
df = pd.DataFrame({"a": [1, 1, 1, 2, 2, 2], "b": range(6)})
df2 = pd.DataFrame({"a": [1, 1, 1, 2, 2, 2, 3], "b": [10 - e for e in range(3)] + list(range(3)) + [10]})
gb = df.groupby("a")
print(gb.corrwith(df2))
#      b   a
# a         
# 1 -1.0 NaN
# 2  1.0 NaN
```

I'm thinking we should deprecate this method.

### PR DESCRIPTION
**Pull request by pedrocariellof**
- [x] closes #57158 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Thanks for the PR! Looks pretty good, a few requests below.

File: pandas/tests/groupby/test_all_methods.py [hunk 1]
```diff
@@ -25,9 +25,15 @@ def test_multiindex_group_all_columns_when_empty(groupby_func):
     gb = df.groupby(["a", "b", "c"], group_keys=False)
     method = getattr(gb, groupby_func)
     args = get_groupby_method_args(groupby_func, df)
-
-    warn = FutureWarning if groupby_func == "fillna" else None
-    warn_msg = "DataFrameGroupBy.fillna is deprecated"
+    if groupby_func == "fillna":
+        warn = FutureWarning
+        warn_msg = "DataFrameGroupBy.fillna is deprecated"
```
[the remark this answers is not in the document]
Comment (pedrocariellof) [reply pandas/tests/groupby/test_all_methods.py:30]: Lines 28-30 repeat in other parts of the code. Should I remove all the instances or just the ones you pointed out?

[the remark this answers is not in the document]
Comment (rhshadrach) [reply pandas/tests/groupby/test_all_methods.py]: Either is fine.

**Comment by pedrocariellof** [comment]
@rhshadrach

File: doc/source/getting_started/install.rst [hunk 2]
```diff
@@ -1,4 +1,4 @@
-.. _install:
```
Comment (rhshadrach) [inline doc/source/getting_started/install.rst:1]: This is still changing

File: doc/source/whatsnew/v3.0.0.rst [hunk 3]
```diff
@@ -171,6 +171,7 @@ Other API changes
 Deprecations
 ~~~~~~~~~~~~
 
+
```
Comment (rhshadrach) [inline doc/source/whatsnew/v3.0.0.rst:174]: Can you remove the newline here.

File: doc/source/whatsnew/v3.0.0.rst [hunk 4]
```diff
@@ -200,6 +201,7 @@ starting with 3.0, so it can be safely removed from your code.
 Other Deprecations
 ^^^^^^^^^^^^^^^^^^
 
+- Deprecated :meth:`Groupby.corrwith` (:issue:`57158`)
```
Comment (rhshadrach) [inline doc/source/whatsnew/v3.0.0.rst:204]:
```suggestion
- Deprecated :meth:`.DataFrameGroupby.corrwith` (:issue:`57158`)
```

**Review by rhshadrach — CHANGES_REQUESTED** [review]
Looking good, a few minor requests.

**Review by rhshadrach — CHANGES_REQUESTED** [review]
lgtm

**Review by rhshadrach — APPROVED** [review]

**Comment by rhshadrach** [comment]
Thanks @pedrocariellof - and for the additional cleanups too!

**Pull request merged**